biogrid/libs/biogrid-simulator/src/lib/biobrain/biobrain-class.ts

```diff
@@ -114,37 +99,36 @@ export class BioBrain implements Brain {
    * @returns @interface SupplyingPath which holds a key value pair of a gridItem requesting mapping to the one which can supplying
    */
   private async chargeLargebatteries(
-    largeBatteries: BioBattery[],
-    solarPanels: SolarPanel[],
-    shortestDistances: { [source: string]: { [node: string]: Path } }
+    shortestDistances: {[source: string]: { [node: string]: Path}}
   ): Promise<SupplyingPath> {
+    const gridItems = this.getGridItems();
+    let largeBatteries: BioBattery[] = gridItems[GRID_ITEM_NAMES.LARGE_BATTERY];
+    let solarPanels: SolarPanel[] = gridItems[GRID_ITEM_NAMES.SOLAR_PANEL];
     // Assuming the large battery is not fully charged
     largeBatteries = largeBatteries.filter((battery) => !battery.isFull());
 
     // Filter the solar panels and remove the ones with the minimum energy or empty
     const solarPanelsFiltered = await this.filterSolarPanelsByEnergyAmount(solarPanels)
 
     // Create an array of the possible energy givers
-    const allEnergyProviders = [...solarPanelsFiltered];
+    const allEnergyProviders = [...solarPanels];
```

Comment:
I see you changed the filter solar panels but did not collect this part. can you insert the right variable which is the filtered solar panels